src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/bolts/FlowSyncHubBolt.java

```diff
@@ -0,0 +1,276 @@
+import org.openkilda.wfm.share.utils.KeyProvider;
+import org.openkilda.wfm.share.zk.ZkStreams;
+import org.openkilda.wfm.share.zk.ZooKeeperBolt;
+import org.openkilda.wfm.topology.flowhs.FlowHsTopology.ComponentId;
+import org.openkilda.wfm.topology.flowhs.FlowHsTopology.Stream;
+import org.openkilda.wfm.topology.flowhs.exception.DuplicateKeyException;
+import org.openkilda.wfm.topology.flowhs.exception.UnknownKeyException;
+import org.openkilda.wfm.topology.flowhs.model.path.FlowPathOperationConfig;
+import org.openkilda.wfm.topology.flowhs.model.path.FlowPathRequest;
+import org.openkilda.wfm.topology.flowhs.model.path.FlowPathResult;
+import org.openkilda.wfm.topology.flowhs.service.FlowSyncCarrier;
+import org.openkilda.wfm.topology.flowhs.service.FlowSyncService;
+import org.openkilda.wfm.topology.flowhs.service.path.FlowPathCarrier;
+import org.openkilda.wfm.topology.flowhs.service.path.FlowPathService;
+import org.openkilda.wfm.topology.utils.MessageKafkaTranslator;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Tuple;
+import org.apache.storm.tuple.Values;
+
+public class FlowSyncHubBolt extends HubBolt implements FlowSyncCarrier, FlowPathCarrier {
+    public static final String BOLT_ID = ComponentId.FLOW_SYNC_HUB.name();
+
+    private final FlowSyncConfig config;
+    private final FlowResourcesConfig flowResourcesConfig;
+
+    private transient FlowSyncService syncService;
+    private transient FlowPathService pathService;
+
+    private transient CarrierContext<String> carrierContext;
+    private LifecycleEvent deferredShutdownEvent;
+
+    public FlowSyncHubBolt(
+            @NonNull FlowSyncConfig config, @NonNull PersistenceManager persistenceManager,
+            @NonNull FlowResourcesConfig flowResourcesConfig) {
+        super(persistenceManager, config);
+        this.config = config;
+        this.flowResourcesConfig = flowResourcesConfig;
+    }
+
+    @Override
+    protected void onRequest(Tuple input) throws PipelineException {
+        FlowSyncRequest request = pullValue(input, FIELD_ID_PAYLOAD, FlowSyncRequest.class);
+        carrierContext.apply(pullKey(input), key -> handleRequest(key, request));
+    }
+
+    @Override
+    protected void onWorkerResponse(Tuple input) throws PipelineException {
+        SpeakerFlowSegmentResponse response = pullValue(input, FIELD_ID_PAYLOAD, SpeakerFlowSegmentResponse.class);
+        carrierContext.apply(pullKey(input), key -> handleWorkerResponse(key, response));
+    }
+
+    @Override
+    protected void onTimeout(String coordinatorKey, Tuple tuple) throws PipelineException {
+        carrierContext.apply(coordinatorKey, key -> syncService.handleTimeout(key));
+    }
+
+    @Override
+    protected boolean deactivate(LifecycleEvent event) {
+        if (syncService.deactivate()) {
```

Comment:
No - pathService do not process any external requests (only requests from syncService) so it can to not implement enable/disable feature and do not need to be called here.

I have added corresponding comment into code here.